src/components/AccordionMenu/index.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import makeClassName from 'classnames';
+
+import { ApplicationState } from '../../reducers';
+import { ConnectedReduxProps } from '../../configureStore';
+import { actions, isExpanded } from '../../reducers/accordionMenu';
+import { AnyReactNode } from '../../typeUtils';
+import styles from './styles.module.scss';
+
+export const makeItemContentId = (title: string) => {
+  const id = title.replace(/[^A-Za-z0-9]+/g, '');
+  if (!id) {
+    throw new Error(`An ID could not be generated from this title: "${title}"`);
+  }
+  return `accordionContent${id}`;
+};
+
+export type PublicItemProps = {
+  children: AnyReactNode;
+  title: string;
+};
+
+export type DefaultItemProps = {
+  expanded: boolean;
+};
+
+type ItemPropsFromState = {
+  expandedFromState: boolean;
```

Comment:
I find this `expanded` and `expandedFromState` a bit confusing. I understand it now after reviewing all the code, but maybe it would be a bit clearer if `expandedFromState` were renamed to just `expanded` (because it represents the "true" expanded state of the item), and if `expanded` were renamed to `expandInitially`, or something like that, to indicate that it is used only to request that an item be initially expanded and after that isn't used for anything.

Reply:
Ah, yes, I struggled with this. Good idea.